### ISSUE TITLE
M5-03 · Processing Outcome, Residual & Terminal Ledger

### ISSUE THREAD

**Issue by catursetyo** [201dev/cirquo#45]
## Dependencies

M5-02

## Summary

Allow a verified Organic Processor to record what a collected recovery batch
became: usable output and unusable residual. This creates the terminal
`PROCESSED` Material Flow Ledger event that M6 uses to derive Recovered and
Residual impact honestly.

## Scope

- Add `recoveryBatches.logOutcome({
  batchId,
  outputType,
  outputWeightGrams,
  residualWeightGrams,
  completedAt?,
  note?
})`.
- Add the outcome form to the owned Processor batch-detail view.
- Transition the batch atomically from `collected → processed`.
- Persist output type, output weight, residual weight, completion time, and
  optional note.
- Compute `processLossGrams` and `conversionRatePercent`.
- Set the parent Rescue Item’s terminal status:
  - `recovered` when output weight is greater than zero;
  - `residual` when no usable output was produced.
- Write exactly one terminal `PROCESSED` event in the same Convex transaction.
- Update Processor and source Merchant views reactively.
- Use Bahasa Indonesia, integer grams, `Intl.NumberFormat('id-ID')`, existing
  shadcn/ui primitives, and Sonner feedback.

## Authorization

The mutation must:

1. Require an authenticated `processor` role.
2. Require a verified Processor profile.
3. Verify `batch.processorId === processor._id`.
4. Require `batch.status === 'collected'`.
5. Require `batch.acceptedWeightGrams` to be present.

No client-side status check is an authorization boundary.

## Validation

- `outputType` must be included in `processor.outputTypes`.
- `outputWeightGrams` and `residualWeightGrams` must be non-negative integer grams.
- Neither value may exceed `acceptedWeightGrams`.
- `outputWeightGrams + residualWeightGrams` may not exceed
  `acceptedWeightGrams`.
- `completedAt` defaults to `Date.now()` and may not be in the future.
- `note` is optional and at most 500 characters.
- A second submission is rejected because `processed` is terminal.

Calculate:

```ts
processLossGrams =
  acceptedWeightGrams - outputWeightGrams - residualWeightGrams

conversionRatePercent =
  Math.round((outputWeightGrams / acceptedWeightGrams) * 1000) / 10
```

`processLossGrams` represents physical loss such as water and CO₂. It is not
counted as either Recovered or Residual.

A residual value of zero is valid, but the UI must request explicit
confirmation because it is unusual.

## Ledger Contract

Write exactly one terminal event in the same transaction as the batch and
Rescue Item updates:

```ts
{
  eventType: 'PROCESSED',
  weightDeltaGrams: -acceptedWeightGrams,
  actorId: processor.ownerId,
  actorRole: 'processor',
  recoveryBatchId: batchId,
  metadata: {
    processorId,
    outputType,
    outputWeightGrams,
    residualWeightGrams,
    processLossGrams,
    conversionRatePercent,
    note,
  },
}
```

Impact attribution must read metadata, not the raw event delta:

```text
Recovered = metadata.outputWeightGrams
Residual  = metadata.residualWeightGrams
Process loss = acceptedWeightGrams - output - residual
```

## UX Requirements

- Make the distinction between usable output, residual, and process loss clear.
- Show the recorded outcome after success; terminal batches cannot be edited.
- Display input validation errors next to their fields.
- Show loading, not-found, access-denied, and mutation-error states.
- Keep the form usable at 375px width with reachable primary actions.
- Do not use `mock-data.ts`.

## Acceptance Criteria

- [ ] Only the assigned verified Processor can log an outcome.
- [ ] An outcome can be logged only after measured intake on a `collected` batch.
- [ ] Unsupported output types and invalid weights are rejected server-side.
- [ ] The mutation prevents creating material beyond the accepted intake weight.
- [ ] A successful outcome persists output, residual, process loss, and completion time.
- [ ] A successful outcome changes the batch to terminal `processed`.
- [ ] The parent Rescue Item becomes `recovered` when output exists, otherwise `residual`.
- [ ] Exactly one `PROCESSED` event with `-acceptedWeightGrams` is written atomically.
- [ ] Recovered and Residual values remain separately derivable from event metadata.
- [ ] Repeated submission cannot duplicate the terminal transition or ledger event.
- [ ] Merchant and Processor see the result reactively.
- [ ] Include a small runnable check for valid outcome weights, process loss, and invalid totals.
- [ ] `bun run lint`, `bun run build`, and `bun scripts/check-ledger.ts` pass.

## Out of Scope

- Intake logging and measured-weight capture — M5-02.
- Processor queue, accept/decline, and M4 routing retries — M5-01 / M4.
- Aggregated impact dashboards — M6.
- Admin variance review, moderation, and manual re-routing — M7.
- Corrections to a completed outcome; they require an append-only compensating
  ledger entry, not mutation of historical data.
